proc-macros/src/rpc_macro.rs

```diff
@@ -295,15 +295,29 @@ impl RpcDescription {
 						));
 					}
 
-					if !matches!(method.sig.output, syn::ReturnType::Default) {
-						return Err(syn::Error::new_spanned(
-							method,
-							"Subscription methods must not return anything; the error must send via subscription via either `SubscriptionSink::reject` or `SubscriptionSink::close`",
-						));
-					}
-
-					if method.sig.asyncness.is_some() {
-						return Err(syn::Error::new_spanned(method, "Subscription methods must not be `async`"));
+					match method.sig.output.clone() {
+						syn::ReturnType::Type(_, ty) => {
+							if let syn::Type::Path(syn::TypePath { path, .. }) = *ty {
+								if let Some(ident) = path.get_ident() {
+									if ident != "SubscriptionResult" && ident != "Result" {
```

Comment:
nit: Could we have an `std::result::Result` here? Should we check just the last segment of the path instead of the whole ident?

Reply:
I think I'm favor of reverting the subscriptions to return result, it's not worth this change  to break everything.

the SubscriptionResult is different for the client and server anyway.